src/client/connect.rs

```diff
@@ -231,6 +248,48 @@ impl ConnectingTcp {
     }
 }
 
+enum HttpConnectorBlockingTaskInner {
+    Dns(oneshot::Execute<dns::Work>)
+}
```

Comment:
Could this inner `enum` go away, and just be a field of `HttpConnectorBlockingTask`?

Reply:
Absolutely! I was setting this up so that we can potentially add different kinds of blocking tasks, but I suppose as long as the API supports it without forcing a breaking change, we can always expand this into an enum later.